### PR TITLE
feat(publish): Custom theme support

### PR DESCRIPTION
**Pull request by SeriousBug**
Docs: https://github.com/dendronhq/dendron-site/pull/492

### CODE REVIEW

**Review by jonathanyeung — CHANGES_REQUESTED** [review]
Were we planning on alpha testing this first with a limited population?  If not, do we want to use the FeatureToaster and TipOfDay to let people know that this option exists?

File: packages/common-all/src/constants/index.ts [hunk 1]
```diff
@@ -44,6 +44,8 @@ export const CONSTANTS = {
   DENDRON_INIT_VERSION: "0.0.0",
   /** Default for the `maxNoteLength` config. */
   DENDRON_DEFAULT_MAX_NOTE_LENGTH: 204800,
+  /** The file containing the custom theme CSS. Located at the workspace root. */
+  CUSTOM_THEME_CSS: "custom.css",
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply packages/common-all/src/constants/index.ts:48]: we can amoeba it - start off by having the same file and splitting it off if users want that

**Review by kevinslin — CHANGES_REQUESTED** [review]
- left some doc comments > https://github.com/dendronhq/dendron-site/pull/492#pullrequestreview-968189217
- can you also confirm that dark theme works?

**Comment by SeriousBug** [comment]
> @kevinslin can you also confirm that dark theme works?

It doesn't actually, the navigation bar shows up in a light theme, which is why I didn't highlight the dark theme support in the docs. I was thinking of solving that in a next iteration so the custom theme support can go out now, should I fix that in this PR instead?

**Comment by kevinslin** [comment]
> It doesn't actually, the navigation bar shows up in a light theme, which is why I didn't highlight the dark theme support in the docs. I was thinking of solving that in a next iteration so the custom theme support can go out now, should I fix that in this PR instead?

ah, nice. would leave it as is then

>     @jonathanyeung alpha testing this first with a limited population

lets do this when we add it add the preview css feature changes

**Review by kevinslin — APPROVED** [review]

File: packages/common-all/src/constants/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanyeung) [reply packages/common-all/src/constants/index.ts:48]: One concern though is that it's harder to rename this file once users have one.  We'd have to do a migration or ask them to do it.

**Comment by jonathanyeung** [comment]
Re: rollout plan, sounds good! - by 'alpha' testing I didn't mean specifically limiting its audience, I was more referring to the treatment of it as a 'germ' stage feature.

**Comment by SeriousBug** [comment]
@jonathanyeung Ah my bad, I must have misunderstood you then. It is an alpha feature, I think the documentation gets this across pretty well. The main thing that makes it alpha is the selectors for which we have no guarantee will be stable.

For the file, I think one way to handle it would be to have this file apply to both preview and publishing. One way to think about the preview is that it's a preview of what the note will look like once it's published, in which case applying the same theme would make more sense. We would then allow overrides in the form of `custom.publish.css` and `custom.preview.css` that apply only to one, for anyone who uses a custom publishing theme but doesn't want that to apply to the preview.

**Pull request merged**